Geode.Angular/src/app/pages/users/users.service.ts

```diff
@@ -0,0 +1,27 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { IUserInfo } from './user-info.dto';
+import { Observable } from 'rxjs';
+import { IFilter } from '../../shared/models/filter.model';
+import { environment } from '../../../environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UsersService {
+  constructor(private http: HttpClient) { }
+
+  getAllUsers(filter: IFilter | null = null): Observable<IUserInfo[]>{
+    let requestUrl = `${environment.apiBase}/user/all`
+    if (filter != null){
+      const queryString = (Object.keys(filter) as Array<keyof IFilter>)
+      .filter(key => !!filter[key])
+      .map(key => `${encodeURIComponent(key)}=${encodeURIComponent(filter[key] as string)}`)
+      .join('&');
+
```

Comment:
You can create some general helper function, that checks if there is a `false` value and removes it if it is. 
Example: 
`
export function filterInvalidValues<T>(obj: T): Partial<T> {
  return Object.keys(obj)
    .filter(key => obj[key] || obj[key] === 0)
    .reduce((result, key) => {
      result[key] = obj[key];
      return result;
    }, {} as Partial<T>);
}
`

And after that you will have a normal object with values. To make a query param, just pass as secondary param. :
```suggestion
const clearedFilter = filterInvalidValues(filter);
return this.http.get<IUserInfo[]>(requestUrl, {params: {...clearedFilter}});
```